lua/ps2/server/sv_pointshopcontroller.lua

```diff
@@ -783,24 +783,28 @@ local function calculateRefundAmounts( itemsToRefund )
     return refundsByPlayer
 end
 
-local function removeSingleItem( itemClass, refund )
+local function removeSingleItem( itemClass, refund, refundCurrentPrice )
     return Pointshop2.DB.DoQuery( Format( [[
         SELECT item.id, item.data, COALESCE(inventories.ownerId, ps2_equipmentslot.ownerId) AS ownerId, ps2_equipmentslot.slotName
         FROM kinv_items AS item
         LEFT JOIN inventories ON inventories.id = item.inventory_id
         LEFT JOIN ps2_equipmentslot ON ps2_equipmentslot.itemId = item.id
         WHERE item.itemClass = %s
-    ]], Pointshop2.DB.SQLStr( itemClass ) ) )
+    ]], Pointshop2.DB.SQLStr( itemClass.name ) ) )
         :Then( function( results )
             results = results or {}
             -- Deserialize purchaseData (this is usually done in LibK but we're querying manually)
             results = LibK._.map( results, function( row )
-                row.purchaseData = util.JSONToTable( row.data ).purchaseData
+                return row.purchaseData = util.JSONToTable( row.data ).purchaseData
             end )
 
             local refundPromise = Promise.Resolve()
             if refund then
-                local refundsByPlayer = calculateRefundAmounts( refund )
+                local currentPrice = false
+				if refundCurrentPrice then
+					currentPrice = table.Inherit(itemClass.static.Price, { points = 0, premiumPoints = 0 })
+				end
+                local refundsByPlayer = calculateRefundAmounts( refund, currentPrice )
                 -- Remove players that get refunded 0 points
                 local toRefund = LibK._( refundsByPlayer ):chain()
                     :entries( )
```

Comment:
Unfortunately I don't have a gmod server to test, would be good to verify this, if `entry[1]` is actually the ownerId the could would try to do e.g. `123.points != 0` which should error -  not sure why it doesn't since you tested this